sdk/python/feast/infra/offline_stores/bigquery.py

```diff
@@ -91,18 +92,28 @@ def get_historical_features(
         expected_join_keys = _get_join_keys(project, feature_views, registry)
 
         if type(entity_df) is str:
-            entity_df_job = client.query(entity_df)
-            # Start the job and get the schema back. We don't need the actual rows.
-            entity_df_result = entity_df_job.result(max_results=0)
+            entity_df_job: QueryJob = client.query(entity_df)
+            # Note: Don't call entity_df_job.result() here, it will lead
+            # to OOM errors. Instead just wait until the job is done.
+            while not entity_df_job.done():
+                time.sleep(1)
 
+            entity_df_sql_table = f"`{entity_df_job.destination.project}.{entity_df_job.destination.dataset_id}.{entity_df_job.destination.table_id}`"
+
+            # Get a single row from this entity table to validate its schema
+            entity_df_limited_job: QueryJob = client.query(
```

Comment:
This seems like a lighter weight approach https://github.com/feast-dev/feast/pull/1495#discussion_r619486530

Reply:
I'm trying this one out now. The previous approach still didn't work for super large datasets. Looks like the anonymous tables were not created when the dataset is large. I'll try creating named table instead.